src/hooks/root/SessionStorage.tsx

```
@@ -0,0 +1,49 @@
+export function SessionStorageProvider({ children, cacheDuration = 4 * 3600 * 1000 }: { children: React.ReactNode; cacheDuration?: number }) {
+  function getStoredValue<T extends object = Any>(key: string, validation?: (value: T | null) => T | never): T | null {
+    const item = JSON.parse(sessionStorage.getItem(key) || 'null')
+    if (!item) return null
+
+    if (!item?.session_expiration || item.session_expiration + cacheDuration < Date.now()) {
+      console.warn('SessionStorageProvider: Item expired, removing from session storage', key)
+      sessionStorage.removeItem(key)
+      return null
+    }
+
+    delete item.session_expiration
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)